src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -140,10 +142,8 @@ private async Task PopulateSnippetCacheAsync(IExpansionManager expansionManager)
 
             // The rest of the process requires being on the UI thread, see the explanation on
             // PopulateSnippetCacheFromExpansionEnumeration for details
-            await Task.Factory.StartNew(() => PopulateSnippetCacheFromExpansionEnumeration(expansionEnumerator),
-                CancellationToken.None,
-                TaskCreationOptions.None,
-                ForegroundTaskScheduler).ConfigureAwait(false);
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+            PopulateSnippetCacheFromExpansionEnumeration(expansionEnumerator);
         }
```

Comment:
intentional that we don't switch back?

Reply:
Yes. One of the following will occur:

* The caller needs to be on the main thread; we stay on that thread and no switch is required on return
* The caller does not need to be on the main thread, and performs its own switch to a background thread